src/flowgger/decoder/rfc3164_decoder.rs

```diff
@@ -63,6 +52,74 @@ struct Pri {
     severity: Option<u8>,
 }
 
+fn decode_rfc_standard(pri: &Pri, msg: &str, line: &str) -> Result<Record, &'static str> {
+    // Decoding "usual" rfc input as advised in the rfc: [<pri>]<datetime> <hostname> <message>
+
+    // The event may have several consecutive spaces as separator
+    let tokens_vec = msg.split_whitespace().collect::<Vec<&str>>();
+
+    // If we have less than 4 tokens, the input can't be valid
+    if tokens_vec.len() > 3 {
+        // Parse the date, the next token is the hostname
+        let (ts, _log_tokens) = parse_date_token(&tokens_vec)?;
+        let _hostname = _log_tokens[0];
+
+        // All that remains is the message that may contain several spaces, so rebuild it
+        let _message = _log_tokens[1..].join(" ");
+
+        let record = Record {
+            ts,
+            hostname: _hostname.to_owned(),
+            facility: pri.facility,
+            severity: pri.severity,
+            appname: None,
+            procid: None,
+            msgid: None,
+            msg: Some(_message.to_owned()),
+            full_msg: Some(line.to_owned()),
+            sd: None,
+        };
+        Ok(record)
+    } else {
+        Err("Malformed RFC3164 standard event: Invalid timestamp or hostname")
+    }
+}
+
+fn decode_rfc_custom(pri: &Pri, msg: &str, line: &str) -> Result<Record, &'static str> {
+    // Decoding custom rfc input formatted as : [<pri>]<hostname>: <datetime>: <message>
```

Comment:
what happned when we find a slightly different one again? Or is this the only real offender in the wild?

Reply:
This is one of device custom log formats, in this case for cisco devices, dates and hostnames makes flowgger compliant with nxos and ios. Most devices seem to comply with the recommanded usage. However there is for sure other formats. I cannot add all existing formats here, but the design update makes it easier to add them on case by case.